packages/applications-service-api/__tests__/integration/submission.test.js

```diff
@@ -0,0 +1,149 @@
+const { request } = require('../__data__/supertest');
+const config = require('../../src/lib/config');
+
+jest.mock('../../src/utils/date-utils');
+jest.mock('../../src/lib/eventClient');
+jest.mock('uuid');
+const { getDate } = require('../../src/utils/date-utils');
+const { sendMessages } = require('../../src/lib/eventClient');
+const uuid = require('uuid');
+
+const submissionRequest = () =>
+	request
+		.post('/api/v1/submissions/BC0110001')
+		.field('name', 'Joe Bloggs')
+		.field('email', 'joe@example.org')
+		.field('interestedParty', 'true')
+		.field('ipReference', '1234567879')
+		.field('deadline', 'Deadline 2')
+		.field('submissionType', 'Comments on LIRs')
+		.field('sensitiveData', 'false')
+		.field('lateSubmission', 'false');
+
+describe('/api/v1/submissions', () => {
+	const mockTime = new Date('2023-12-30T11:06:13.245Z');
+	const mockUuid = 'eb43f2de-cd51-4f95-a7a3-e04082bdcd8b';
+
+	beforeEach(() => {
+		config.backOfficeIntegration.submissions.postSubmission.caseReferences = ['BC0110001'];
+		uuid.v4.mockReturnValue(mockUuid);
+	});
+
+	describe('POST /:caseReference', () => {
+		describe('Back Office case reference', () => {
+			const requestSubmissionId = 'BC0110001-101023134030108';
+			const mockGeneratedSubmissionId = 'BC0110001-301223110613245';
+			const expectedMessage = {
+				blobGuid: mockUuid,
+				caseReference: 'BC0110001',
+				deadline: 'Deadline 2',
+				email: 'joe@example.org',
+				interestedParty: true,
+				interestedPartyReference: '1234567879',
+				lateSubmission: false,
+				name: 'Joe Bloggs',
+				sensitiveData: false,
+				submissionType: 'Comments on LIRs'
+			};
+
+			describe('submission with written representation', () => {
+				describe('request with submissionId provided', () => {
+					it('uploads pdf of representation, publishes message, and returns submissionId', async () => {
```

Comment:
This might be partly preference but I think it would be better for the tests to be isolated (not have "and" in test blocks) - have the `sendMessage` assertion in a different block and the returns `submissionId` in its own one. It makes it clear what is failing if something starts breaking. You can just throw the `submissionRequest` in a beforeEach.

Reply:
It's a good shout, and happy to go with a different convention. I usually write the tests quite "fat" with lots of expectations, mainly to reduce repetition of the arrange/act parts but as you say - that could be handled in a `beforeEach`. They certainly read better when you don't have lots of text in the `it` and break it down into a more nested structure. I'll be updating these tests in my next ticket so will do some restructuring in the next PR